core/src/test/java/org/kiwix/kiwixmobile/core/page/viewmodel/PageViewModelTest.kt

```diff
@@ -0,0 +1,171 @@
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.verify
+import io.reactivex.plugins.RxJavaPlugins
+import io.reactivex.processors.PublishProcessor
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.schedulers.TestScheduler
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.kiwix.kiwixmobile.core.dao.PageDao
+import org.kiwix.kiwixmobile.core.page.PageImpl
+import org.kiwix.kiwixmobile.core.page.adapter.Page
+import org.kiwix.kiwixmobile.core.page.pageState
+import org.kiwix.kiwixmobile.core.page.viewmodel.Action.Exit
+import org.kiwix.kiwixmobile.core.page.viewmodel.Action.ExitActionModeMenu
+import org.kiwix.kiwixmobile.core.page.viewmodel.Action.Filter
+import org.kiwix.kiwixmobile.core.page.viewmodel.Action.OnItemClick
+import org.kiwix.kiwixmobile.core.page.viewmodel.Action.OnItemLongClick
+import org.kiwix.kiwixmobile.core.page.viewmodel.Action.UpdatePages
+import org.kiwix.kiwixmobile.core.page.viewmodel.Action.UserClickedDeleteButton
+import org.kiwix.kiwixmobile.core.page.viewmodel.Action.UserClickedDeleteSelectedPages
+import org.kiwix.kiwixmobile.core.page.viewmodel.Action.UserClickedShowAllToggle
+import org.kiwix.kiwixmobile.core.page.viewmodel.effects.OpenPage
+import org.kiwix.kiwixmobile.core.reader.ZimReaderContainer
+import org.kiwix.kiwixmobile.core.search.viewmodel.effects.Finish
+import org.kiwix.kiwixmobile.core.utils.SharedPreferenceUtil
+import org.kiwix.sharedFunctions.InstantExecutorExtension
+import org.kiwix.sharedFunctions.setScheduler
+
+@ExtendWith(InstantExecutorExtension::class)
+internal class PageViewModelTest {
+  private val pageDao: PageDao = mockk()
+  private val zimReaderContainer: ZimReaderContainer = mockk()
+  private val sharedPreferenceUtil: SharedPreferenceUtil = mockk()
+
+  private lateinit var viewModel: TestablePageViewModel
+  private val testScheduler = TestScheduler()
+  private val itemsFromDb: PublishProcessor<List<Page>> =
+    PublishProcessor.create()
+
+  init {
+    setScheduler(testScheduler)
+    RxJavaPlugins.setIoSchedulerHandler { Schedulers.trampoline() }
+  }
+
+  @BeforeEach
+  fun init() {
+    clearAllMocks()
+    every { zimReaderContainer.id } returns "id"
+    every { zimReaderContainer.name } returns "zimName"
+    every { sharedPreferenceUtil.showHistoryAllBooks } returns true
+    every { pageDao.pages() } returns itemsFromDb
+    viewModel = TestablePageViewModel(zimReaderContainer, sharedPreferenceUtil, pageDao)
+  }
+
+  @Test
+  fun `initial state is Initialising`() {
+    viewModel.state.test().assertValue(pageState())
+  }
+
+  @Test
+  fun `Exit finishes activity`() {
+    viewModel.effects.test().also { viewModel.actions.offer(Exit) }.assertValue(Finish)
+    viewModel.state.test().assertValue(pageState())
+  }
+
+  @Test
+  fun `ExitActionModeMenu calls deslectAllPages`() {
+    mockkObject(viewModel)
```

Comment:
This seems wrong, `mockkObject` is for mocking `object`s. What you could do is use `spy` or seeing as you own TestableViewModel, you could pass in a mockk() of `PageViewModel` and in the overriden methods delegate the function to the mock's functions and then you would verify the mockViewModel had its functions called